### PR TITLE
The LEVYING! (Garrison Class Selection - As Promised)

### PR DESCRIPTION
**Pull request by Stutternov**
## About The Pull Request

Basically adds a garrison 'class' selection system, this way we don't have to have garrison archer and garrison split into separate roles; as well as allowing specialization loadouts for people to choose a sort of 'class' that they specifically like. I've seen often guards be a bit upset they roll weird and can't use their favorite playstyle. So, here, this should alleviate that! 

This acts like an Adventurer, giving you the round pop-up 'select your class' list but of only guard classes! Yippie!

### The Guard Types

Four guard classes have been added as selectable under the garrison job-type. They are as follows:

- **Footsman**, the most basic of guard types. Footsmen basically use what guards prior had with some small changes, focusing on axes, maces, and shields.
- **Pikesman**, a guard with slightly more strength by one but less speed by one in turn. Pikesmen start with either a spear or billhook and sport a slightly different look via their gear.
- **Archer**, unchanged from garrison archer but merged into the role. Exact same as before with the RnG removed; now you roll 1 crafting, 1 tanning, 1 sword. (As opposed to: 1-2 crafting, 1-2 tanning, 0-1 sword. Standardized.)
- **Arcane Enforcer**, a guard focusing on _magic_ usage, mixing together a few combat stats and lesser non-lethal spell types. Limited to 2 slots these guards are still push-overs compared to others in combat but serve a massive support use, and can hold themselves to a minor extent; though are out-preformed by nearly any combat-dedicated job.

## Why It's Good For The Game

Adds variety to the Garrison roles. This type of PR has been done on other servers with pretty positive reception. It allows people to play into specific types of roles they like and act as a character with specialties rather than expecting to be well-rounded.

This was requested by staff and some players, and I don't see why not. My only need is feedback; as roles, such as the Arcane role, is a bit up in the air as its balance is hard to determine without actual playtesting with players vs players. While they are actually weaker spells-wise than any other mage role, they do have some bonus stats (mostly in speed) over those roles, so the balancing might be alright..?

## Pre-Merge Checklist

- [ ] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [ ] You documented all of your changes.

### CODE REVIEW

**Comment by Stutternov** [comment]
Note: This is drafted while I receive feedback from the staff members who requested this, coder-staff for balance helping, and community feedback on the general idea.

**Comment by sanshoom** [comment]
call arcane enforcers warlocks :wise_tree:

File: code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm [hunk 1]
```diff
@@ -41,42 +43,181 @@
...
+
+	category_tags = list(CTAG_GARRISON)
+
+/datum/outfit/job/roguetown/guardsman/pikesman/pre_equip(mob/living/carbon/human/H)
+	..()
+	//Armor for class
+	armor = /obj/item/clothing/suit/roguetown/armor/chainmail
+	neck = /obj/item/clothing/neck/roguetown/gorget
+	head = /obj/item/clothing/head/roguetown/helmet/kettle
+	if(prob(50))
```
Comment (ThePainkiller) [inline code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm:116]: Lower prob please, like 33% or something. Billhook is a really good weapon, and a good justification for them to purchase one from a smith.

File: code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm [hunk 2]
```diff
@@ -41,42 +43,181 @@
...
+	H.mind.adjust_skillrank(/datum/skill/combat/bows, 5, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/combat/crossbows, 4, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/combat/wrestling, 3, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/combat/unarmed, 2, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/misc/swimming, 3, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/misc/reading, 1, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/misc/climbing, 3, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/misc/athletics, 3, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/misc/lockpicking, 2, TRUE)
+	H.mind.adjust_skillrank(/datum/skill/combat/swords, 1, TRUE) // For rapier use
```
Comment (ThePainkiller) [inline code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm:169]:
```suggestion
	H.mind.adjust_skillrank(/datum/skill/combat/swords, 2, TRUE) // For rapier use
```
Considering last admin meeting with a consensus to give non-combat weapons a standardized level 2 on weapon skills, up this to swords 2 for consistency.

**Review by ThePainkiller — CHANGES_REQUESTED** [review]

**Comment by NPC1314** [comment]
Good concept.
Skills should be aligned with #538
Perhaps add a root pre_equip for the skills in common to make it easier to overview?
Mage guard is not a good idea IMO, should be kept to court mage, occasional wandering mage, inflation
of number of characters with the same special talents makes them a lot less interesting, also balance problem.

**Comment by 1N7Slayer** [comment]
I like the arcane enforcer idea. Roguetown has been too magic-lite for awhile, so I'd like to see this server stand out from the others by comparison.

File: code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm [hunk 3]
```diff
@@ -41,42 +43,181 @@
...
+	..()
+	//Armor for class
+	armor = /obj/item/clothing/suit/roguetown/armor/leather/hide
+	beltr = /obj/item/reagent_containers/glass/bottle/rogue/manapot		//Fortnite chugjug
+	backr = /obj/item/rogueweapon/woodstaff								//Stinky-uggy wizard staff. Go do kung-fu.
+	head = /obj/item/clothing/head/roguetown/helmet/kettle
+	neck = /obj/item/clothing/neck/roguetown/coif						//Screw you, magic-man. You get a worse coif!
+
+	//Stats for class
+	H.mind.adjust_skillrank(/datum/skill/combat/knives, 3, TRUE)
```
Comment (Valkrae) [inline code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm:198]: personally, i think any weapon skill above 3 might be too much.

Comment (ThePainkiller) [reply code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm:198]: But this is exactly 3, as expected of every town guard as per our admin meeting...

Comment (Valkrae) [reply code/modules/jobs/job_types/roguetown/garrison/garrisonguard.dm:198]: its a mage, though. not a direct combat class. it's supposed to be a support element, that's why I'm saying any weaponskill above 3. Sorry, it doesn't show it in the review, but this particular 3 knives is for the Arcane Enforcer class.

**Comment by Mojamble** [comment]
I understand why people don't like the idea of guard mages, but at the same time, is a taser guard REALLY such a bad idea? Would lead to more non-lethal take downs, plus they'll get crumpled in a real fight.

**Comment by Optimism333** [comment]
## Review ##
### Footman ###
image

- Would suggest giving this class skilled (3) shields at minimum, and also skilled (3) swords, so they can use their board with a sword, axe, or mace depending on preference.
- I don't think they need polearm skill (since that is the focus of the Pikeman class), or riding skill
- I would also suggest slightly adjusting their stats to something like this: +1 strength, +2 endurance, +1 constitution. Dropping that point of perception to give them a bit more stamina for those drawn out close quarters fights

### Pikeman ###
image

- Pikeman stats are probably alright as is, would just remove the +1 to perception as they do not need it
- I would suggest giving them skilled (3) in polearms and axes/maces (to use the grand mace, and potentially the long axes that will come with Dromkii's prs), but no shield skill

### Garrison Archer ###
image

- Would suggest dropping knife skill from expert (4) to skilled (3), to match with the baseline for common combat classes. Other than that, this one seems fine

### Garrison Arcane Enforcer ###
image

- I would suggest just giving them skilled (3) polearms so they can use their staff for wizard-fu, and maybe average (2) knives
- If they're apprentice level magic users, I would reduce their Intelligence to +2, and perhaps reduce their -2 in strength to -1
- This class is a bit of a balancing eniga, to be frank. But hopefully we can figure out what fits best in time

**Review by DovaKitty — CHANGES_REQUESTED** [review]
These lines of code are not implemented in garrisonguard.dm and will need to be added for subclass selection to function as mercenary does. Currently this PR spawns guards in with just their base gear with no subclass select.
image

**Comment by Stutternov** [comment]
Note: This PR is waiting for nets to go in fully. I have an idea......

**Comment by ThePainkiller** [comment]
> Note: This PR is waiting for nets to go in fully. I have an idea......

Nets are fullmerged, nerd. Go ham.

**Comment by NPC1314** [comment]
SK divorded 544 and married the younger version in 809. Time to get old yellered.

**Pull request closed without merging**